src/apis/WikipediaApi.py

```diff
@@ -36,17 +47,11 @@ def get_summaries(self, titles):
         :return: A dict of titles to summary strings
         """
         ts = time()
-        queue = Queue()
-        summaries = dict()
-        for x in range(self.worker_count):
-            worker = self.WikipediaSummaryWorker(queue, self.api_root, self.headers, summaries, self.cache)
-            worker.daemon = True
-            worker.start()
         for title in titles:
-            queue.put(title)
-        queue.join()
+            self.worker_queue.put(title)
+        self.worker_queue.join()
         log.debug("Took {} seconds to fetch {} summaries".format(time() - ts, len(titles)))
-        return summaries
+        return self.summaries
```

Comment:
Do you ever reset the summaries dict or does it just keep accumulating results? Will this be a memory issue if it grows too large?